tests/system/Database/BaseQueryTest.php

```diff
@@ -448,7 +605,6 @@ public function testHighlightQueryKeywords($expected, $sql)
     {
         $query = new Query($this->db);
         $query->setQuery($sql);
-        $query->getQuery();
```

Comment:
Why do you remove it?

Reply:
This had been noticed previously, see [above](https://github.com/codeigniter4/CodeIgniter4/pull/5127#pullrequestreview-781371440).

It's because `getQuery()` can now be called multiple times without error, I could remove the ugly hack [here](https://github.com/codeigniter4/CodeIgniter4/pull/5127/files#diff-33f862bc94ae2386d78473bf248d6bf3f04c342f3044ce52870dd7ddf7b0c37fR411). That hack was in place so that `getQuery()` isn't called an additional time.

But… it required that `getQuery()` had been be called beforehand, so that `finalQueryString` is set, as `compileBinds()` needed it to be set beforehand… hence the "artificial" `getQuery()` here, just so that `finalQueryString` is set.

Reply:
I see.